### PR TITLE
Use Dual workflow for Confluence OAuth

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR implements dual mode for Confluence OAuth. This change is gated behind a FF.

Tested locally.

### CODE REVIEW

File: connectors/src/connectors/confluence/lib/confluence_api.ts [hunk 1]
```diff
@@ -19,6 +23,35 @@ export async function getConfluenceCloudInformation(accessToken: string) {
   }
 }
 
+export async function getConfluenceAccessToken(connectionId: string) {
+  if (isDualUseOAuthConnectionId(connectionId)) {
+    const tokRes = await getOAuthConnectionAccessToken({
```
Comment (spolu) [inline connectors/src/connectors/confluence/lib/confluence_api.ts:28]: You want to use -WithThrow here :+1:

**Review by fontanierh — APPROVED** [review]

File: connectors/src/connectors/confluence/index.ts [hunk 2]
```diff
@@ -64,11 +58,8 @@ export class ConfluenceConnectorManager extends BaseConnectorManager<null> {
     connectionId: string;
   }): Promise<Result<string, Error>> {
     const nangoConnectionId = connectionId;
-    const confluenceAccessToken = await getAccessTokenFromNango({
-      connectionId: nangoConnectionId,
-      integrationId: getRequiredNangoConfluenceConnectorId(),
-      useCache: false,
-    });
+    const confluenceAccessToken =
+      await getConfluenceAccessToken(nangoConnectionId);
```
Comment (spolu) [inline connectors/src/connectors/confluence/index.ts:62]: But you don't want to throw here

File: connectors/src/connectors/confluence/index.ts [hunk 3]
```diff
@@ -139,13 +130,8 @@ export class ConfluenceConnectorManager extends BaseConnectorManager<null> {
...
 
-      const newConnection = await getConnectionFromNango({
-        connectionId,
-        integrationId: getRequiredNangoConfluenceConnectorId(),
-        refreshToken: false,
-      });
-
-      const confluenceAccessToken = newConnection?.credentials?.access_token;
+      const confluenceAccessToken =
+        await getConfluenceAccessToken(connectionId);
```
Comment (spolu) [inline connectors/src/connectors/confluence/index.ts:134]: Or here

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**